src/core/ddsi/src/q_entity.c

```diff
@@ -385,7 +385,9 @@ static void local_reader_ary_remove (struct local_reader_ary *x, struct reader *
   for (i = 0; i < x->n_readers; i++)
     if (x->rdary[i] == rd)
       break;
-  assert (i < x->n_readers);
+  if (i >= x->n_readers) {
+    return; // rd not found, nothing to do
```

Comment:
Could also be done with #ifdefs at the call site, but I think it is safer this way.